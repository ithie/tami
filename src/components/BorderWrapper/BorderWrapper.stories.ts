import type { Meta, StoryObj } from '@storybook/vue3'
import BorderWrapper from '@/components/BorderWrapper/BorderWrapper.vue'

const meta = {
  title: 'Components/BorderWrapper',
  component: BorderWrapper,
  render: () => ({
    components: { BorderWrapper },
    template: '<BorderWrapper />'
  }),
  parameters: {},
  args: {},
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {}
