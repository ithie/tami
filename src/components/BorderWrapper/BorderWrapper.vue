<template>
  <div class="border-all">
    <slot />
  </div>
</template>

<script lang="ts" setup></script>

<style lang="scss">
.border-all {
  max-width: 300px;
  background-color: lightblue;
  padding: 15px;
}
</style>
