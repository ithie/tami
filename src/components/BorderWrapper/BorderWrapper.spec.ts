import { describe, test, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BorderWrapper from './BorderWrapper.vue'

describe('components/BorderWrapper', () => {
  test('should render', async () => {
    const wrapper = mount(BorderWrapper)

    expect(wrapper.element).toMatchSnapshot()
  })
  test('should render without child', async () => {
    const wrapper = mount({
      components: { BorderWrapper },
      template: '<BorderWrapper><div>Child Element</div></BorderWrapper>'
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
