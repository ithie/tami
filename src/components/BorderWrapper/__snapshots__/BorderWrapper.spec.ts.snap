// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/BorderWrapper > should render 1`] = `
<div
  class="border-all"
>
  
  
</div>
`;

exports[`components/BorderWrapper > should render without child 1`] = `
<div
  class="border-all"
>
  
  <div>
    Child Element
  </div>
  
</div>
`;
